euphonic/plot.py

```diff
@@ -131,95 +139,144 @@ def plot_1d(spectra, title='', x_label='', y_label='', y_min=None, labels=[],
     return fig
 
 
-def plot_2d(spectrum, vmin=None, vmax=None, ratio=None,
-            cmap='viridis', title='', x_label='', y_label=''):
+def _plot_2d_core(spectrum: Spectrum2D, ax: Axes,
+                  cmap: Union[str, mpl.colors.Colormap] = 'viridis',
+                  interpolation: str = 'nearest',
+                  norm: Optional[Normalize] = None,
+                  ) -> NonUniformImage:
+    """Plot Spectrum2D object to Axes
+
+    Parameters
+    ----------
+    spectrum
+        2D data object for plotting as NonUniformImage. The x_tick_labels
+        attribute will be used to mark labelled points.
+    ax
+        Matplotlib axes to which image will be drawn
+    cmap
+        Matplotlib colormap or registered colormap name
+    interpolation
+        Interpolation method: 'nearest' or 'bilinear' for a pixellated or
+        smooth result
+    norm
+        Matplotlib normalization object; set this in order to ensure separate
+        plots are on the same colour scale.
+
+    """
+    x_unit = spectrum.x_data_unit
+    y_unit = spectrum.y_data_unit
+    z_unit = spectrum.z_data_unit
+
+    x_bins = spectrum._get_bin_edges('x').to(x_unit).magnitude
+    y_bins = spectrum._get_bin_edges('y').to(y_unit).magnitude
+
+    image = NonUniformImage(ax, interpolation=interpolation,
+                            extent=(min(x_bins), max(x_bins),
+                                    min(y_bins), max(y_bins)),
+                            cmap=cmap)
+    if norm is not None:
+        image.set_norm(norm)
+
+    image.set_data(spectrum._get_bin_centres('x').to(x_unit).magnitude,
+                   spectrum._get_bin_centres('y').to(y_unit).magnitude,
+                   spectrum.z_data.to(z_unit).magnitude.T)
+    ax.images.append(image)
+    ax.set_xlim(min(x_bins), max(x_bins))
+    ax.set_ylim(min(y_bins), max(y_bins))
+
+    _set_x_tick_labels(ax, spectrum.x_tick_labels, spectrum.x_data)
+
+    return image
+
+
+def plot_2d(spectra: Union[Spectrum2D, Sequence[Spectrum2D]],
```

Comment:
The plot functions need to be tested and the docs need to be updated for the new interface (it now just returns a figure). This doesn't have to be done until we've finalised everything in the feature branch but just reminding myself that it needs to be done.

Reply:
I really want to put this off but probably shouldn't!